### DISCUSSION TITLE
Trying to query data related by foreign key, getting empty arrays as a result.

### DISCUSSION THREAD

**Discussion by lumenwrites** [supabase/supabase#13334] [Questions]
Hi! 

I have a `Profile` table that lists basic user info (like their emails), and a `ProfileCourse` table that lists the video courses the user has purchased (this table has foreign keys to `Profile.id` and `Course.id`):
<img width="1071" alt="Screen Shot 2023-03-31 at 07 40 29" src="https://user-images.githubusercontent.com/4761235/229043137-bfdccb46-69cb-442c-9efc-0d331bc10476.png">

My goal is to query a list of all the `Course`s owned by the `Profile`. I'm trying to do it like this:
```
      const { data: profile } = await supabaseClient
        .from('Profile')
        .select('id, username, email, stripeCustomerId, ProfileCourse(*)')
        .eq('id', helpersUser?.id)
        .single()
```

This successfully returns the data like `email` and `username`, but the `ProfileCourse` shows me an empty array (even though the user owns the courses).

Can you help me to understand why, and what can I do to fix this?

Edit:
Oh, it looks like an RLS issue, if I disable RLS on `ProfileCourse` table it's working. Can you help me figure out how to write an RLS policy that would enable the users to read the courses they own from the `ProfileCourse` table?

**Comment by GaryAustin1** [comment]
Is profileId the same as the user UUID? 
If so you can just use auth.uid() = "profileId".
If not you would need to use something like:
```
auth.uid() in (
      select user_id from "Profiles"
      where "Profiles.profileId" = "ProfileCourse.profileId")
```
Note you really should not use capital letters for names in Postgres unless forced to as it is a pain to keep everything in double quotes.
